web/modules/custom/custom_weather/src/Service/UserCityHandler.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Drupal\custom_weather\Service;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Session\AccountProxyInterface;
+
+/**
+ * Determines the city for displaying weather.
```

Comment:
Is it?
It at least saves the city as well.
So not really "determining" from nothing.